apps/site/assets/ts/trip-planner/__tests__/TripPlannerPageTest.tsx

```diff
@@ -0,0 +1,201 @@
+import React from "react";
+import renderer, { act } from "react-test-renderer";
+import {
+  MapData,
+  MapMarker as Marker
+} from "../../leaflet/components/__mapdata";
+import { createReactRoot } from "../../app/helpers/testUtils";
+import TripPlannerPage, {
+  Action,
+  RemoveEvent,
+  State,
+  UpdateEvent,
+  onRemove,
+  onUpdate,
+  reducer
+} from "../components/TripPlannerPage";
+const mapData: MapData = {
+  default_center: { latitude: 0, longitude: 0 },
+  height: 500,
+  markers: [],
+  polylines: [],
+  tile_server_url: "",
+  width: 630,
+  zoom: 14
+};
+
+describe("TripPlannerPage", () => {
+  it("renders", () => {
+    createReactRoot();
+
+    const tree = renderer
+      .create(<TripPlannerPage mapData={mapData} />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("registers event listeners", () => {
+    const cbHash: { [event: string]: Function } = {};
+    document.addEventListener = (name: string, cb: EventListener) => {
+      cbHash[name] = cb;
+    };
+    renderer
+      .create(<TripPlannerPage mapData={mapData} />)
+      // update triggers useEffect to run
+      .update(<TripPlannerPage mapData={mapData} />);
```

Comment:
This `update()` can be removed, and the expect can be moved to line 66 without effecting coverage since the code below will cause a dispatch and state change.

Reply:
I tried removing both the `.update` and the `act` lines, and that key test still passed, so I don't think it's the code below that's triggering it. Which makes sense, since those tests are fetching their callbacks from the hash.

I've updated to add an expect that the callbacks don't throw, so there's at least some testing going on. I confirmed that it does fail if the callback is given a bad argument.